### ISSUE TITLE
import torch.autograd.gradcheck import zero_gradients is unsuccessful

### ISSUE THREAD

**Issue by pstarconsult** [Trusted-AI/adversarial-robustness-toolbox#1911]
I found the fix for this! Just comment out or remove the import below in the first cell of this notebook. The import is never used. I was able to run the entire notebook successfully:

In the first cell of imperceptible_attack_on_tabular_data.ipynb, comment out or remove this import statement:

from torch.autograd.gradcheck import zero_gradient

**Describe the bug**
While trying to run: 

(https://github.com/Trusted-AI/adversarial-robustness-toolbox/tree/main/notebooks)/imperceptible_attack_on_tabular_data.ipynb, 

I get the error:

---> 18 from torch.autograd.gradcheck import zero_gradients
ImportError: cannot import name 'zero_gradients' from 'torch.autograd.gradcheck' (/home/paul/Desktop/ART-AML/art-aml/lib/python3.10/site-packages/torch/autograd/gradcheck.py)

**To Reproduce**
1. Load imperceptible_attack_on_tabular_data.ipynb
2. Run the first cell

**Expected behavior**
The cell should import torch.autograd.gradcheck import zero_gradients when the first cell is run. 

Please see https://stackoverflow.com/questions/68419612/imorting-zero-gradients-from-torch-autograd-gradcheck where it is indicated that "This seems like it's using a very old version of PyTorch, the function itself is not available anymore".

Note that I did perform 'pip install adversarial-robustness-toolbox[pytorch]', got the error and then ran 'pip install --upgrade torch' and I still get the error.  

Also see: https://splunktool.com/imorting-zerogradients-from-torchautogradgradcheck where it indicates this error occurs after moving from PyTorch version 1.8 to 1.9.0. I ran 'pip list | grep torch' and it indicates 1.13.0. 

**Screenshots**
N/A

**System information (please complete the following information):**
 - OS: Ubuntu 22.02
 - Python version: 3.10.6
 - ART version or commit number - latest, installed Nov 11, 2022
 - PyTorch

**Comment by beat-buesser** [comment]
Hi @pstarconsult Thank you very much for your interest in ART and opening this issue! I think you are right, we can remove the import of `zero_gradients` because it does not seem to be used later in the notebook.

If your time allows, would you be interested in opening a pull request targeting branch `main`?

**Comment by pstarconsult** [comment]
@beat-buesser I would be happy to do a pull request targeting main. It may take me a day or so but I will get on this. I am also willing to take on more issues that get filed if you need that. Just let me know. In the meantime, watch for the pull. Thank you!

**Comment by pstarconsult** [comment]
@beat-buesser I have not forgotten about this. I am working out some changes to our repos. I have the fork and commit ready for a push. I will submit soon. Thank you for your patience!

**Comment by beat-buesser** [comment]
@pstarconsult Sounds great and no rush!

**Comment by pstarconsult** [comment]
@beat-buesser I sent pull request but let me know if it passes muster. This is my first contribution so I am open to input. It was a very simple issue so I am not sure how much there is to critique. I would enjoy contributing in more ways. Thank you!

**Comment by pstarconsult** [comment]
@beat-buesser Thank you for the input! I will try your git push a little later today (when I have my lab machine booted up) but I see these results in #1927:

11 failing, 21 successful, 1 action required, 3 expected, and 1 neutral checks

I am a little rusty here but these look like other tests you run after a bug is checked in (?). :)

I will catch you on Slack for some of the other discussion such as how to contribute. 

Thank you again!

**Comment by beat-buesser** [comment]
Hi @pstarconsult Great! I have responded on the tests in the PR. On Slack we have channels #random and #bugs for open discussions or direct messages.

**Comment by pstarconsult** [comment]
I have pushed the latest changes to this notebook, specifically removing the import that was commented out.

**Comment by pstarconsult** [comment]
These are notes for the latest (final?) commit of this notebook on two issues. 

1. To remove the aforementioned import that was useless and caused an import error. 
2. As series of benign errors were generated from cells 22 and 23 resulted in dozens of pointless lines of output. For some reason, the underlying code indicates that X (training data) was trained without feature names. Adding '.values.' to X_train_iris (i.e. X_train_iris.values) fixed this. See the error below and what the fix looks like in one example from the code:

Error: /home/paul/Desktop/art-aml-fork/adversarial-robustness-toolbox/art-aml/lib/python3.10/site-packages/sklearn/base.py:443: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  warnings.warn(

Fix: log_regression_clf_iris.fit(X_train_iris.values, y_train_iris)

There is one remaining warning that persists for these two cells and I could not figure out how to fix them given the time I had to work on this bug. I think this has to do with how underlying code is indexing - perhaps it is an issue that one function (somewhere) is indexing at zero while another is indexing at 1. Not sure. Suffice it to say that I was squeamish about making any changes to code that was not in the notebook itself. I think these two errors are livable (?).  

Note that I initially tried to suppress the specific warnings but that did not seem to work so I fixed them instead. 

3. A final note that it is possible for the output from the cell to throw off the html for the output resulting in a JSON file that will not be notebook-compliant.
